ledger/participant-integration-api/src/main/scala/platform/apiserver/execution/StoreBackedCommandExecutor.scala

```diff
@@ -56,7 +57,13 @@ private[apiserver] final class StoreBackedCommandExecutor(
   ): Future[Either[ErrorCause, CommandExecutionResult]] = {
     val interpretationTimeNanos = new AtomicLong(0L)
     val start = System.nanoTime()
+    val coids = commands.commands.commands.toSeq.foldLeft(Set.empty[Value.ContractId]) {
+      case (acc, com.daml.lf.command.ApiCommand.Exercise(_, coid, _, argument)) =>
+        argument.collectCids(acc) + coid
+      case (acc, _) => acc
+    }
     for {
+      _ <- contractStore.prefetchContracts(coids.toSeq)
```

Comment:
looks like we let the size of the commands dictate the batch retrieval, which is probably okay, but to be bullet proof here probably an upper bound would need to be defined

Reply:
Generally, I don't think this is really needed, but we definitively can group the query in chunks.